language/vm/src/file_format.rs

```diff
@@ -1365,23 +1368,130 @@ impl CompiledScript {
     ///
     /// If a `CompiledScript` has been bounds checked, the corresponding `CompiledModule` can be
     /// assumed to pass the bounds checker as well.
-    pub fn into_module(self) -> CompiledModule {
-        CompiledModule(self.0.into_module())
+    #[allow(deprecated)]
+    pub fn into_module(self) -> (ScriptConversionInfo, CompiledModule) {
+        let (info, m) = self.0.into_module();
+        (info, CompiledModule(m))
     }
 }
 
+pub struct ScriptConversionInfo {
+    // If a dummy address was added.
+    dummy_addr: bool,
+    // If an empty signature was added.
+    dummy_sig: bool,
+    // If the <SELF> identifier was added.
+    self_ident: bool,
+    // If a dummy self module handle was added.
+    self_module_handle: bool,
```

Comment:
nit, maybe all of these boolean flag names should be in the form of predicates? 
`added_dummy_addr`
`added_dummy_sig`
etc

I find it very confusing when the name of something doesn't really line up with the type. For example I would expect `dummy_addr` to have type `AccountAddress` or `AddressIdentifierIndex`

Reply:
That sounds much better! Thanks for suggesting this!